### DISCUSSION TITLE
Nesting for event pages?

### DISCUSSION THREAD

**Discussion by bittlingmayer** [machinetranslate/machinetranslate.org#291] [General]
We decided not to make event instance pages (e.g. WMT22) as "children" of the event class pages (WMT).

But maybe that was wrong?

I see 2 benefits to making them instances:
- We'll get breadcrumbs automatically.  (It's a bit odd that, right now, the WMT22 page doesn't link to the WMT page.)
- The list will in the sidebar will be much easier to navigate.

I see 1 drawback:
- A random small event will be at a higher-level in the sidebar than e.g. WMT22.

**Comment by cefoo** [comment]
Yes, I totally agree.

I'd like to do that with AMTA, EAMT, AAMT as well, but it can be trickier. All of these have an entry under `Organisations`, and that makes the current `WMT` display sort of imbalanced (it's not exactly the same, but could be comparable).

The fix I think is easier right now, and which is not my favorite, is: we could move each organisation's description to Event's higher level, and then nest each organisation's events... Or maybe we could move `wmt.md` to `Organisations`, which is not exactly where it is supposed to be, but would solve its odd location right now.

The thing is that:

- `Organisations` would be almost empty if we do this.
- `Events` would contain information that does not belong there.

Either way, we can nest events in a structure similar to this:

---
## Events
- AMTA
  - AMTA 2020
  - AMTA 2021
  - AMTA 2022
- EAMT
  - EAMT 2020
  - EAMT 2021
  - EAMT 2022
- WMT
  - WMT20
  - WMT21
  - WMT22
- Further events
  -  ...
--- 

What would constitute a smaller or random event? Are those events the ones that are not recurrent?

What if we separe them among `Conferences` and `Workshops`?

**Comment by bittlingmayer** [comment]
You're right, they're also orgs, and WMT could be one too.

I realised we can't just make structure in the events sidebar, because the event series pages will shadow the paths for the orgs pages, like /amta.

So I see a few choices:
- Just make events in reverse chronological order in the sidebar.
- Use full names like /association-for-machine-translation-in-the-americas for the organisations, and use the abbreviations like /amta for the event series

I think either way we'll need to add `organisation:` to the front matter of event instances.  And then we'll be able to autogenerate the list of all events in a series i.e. from the same organisation.

**Comment by cefoo** [comment]
> Just make events in reverse chronological order in the sidebar.

I agree, the sidebar should reflect the order in which events appear in the Events table.
I think we should do this regardless of whether we want to nest events or not.

> Use full names like /association-for-machine-translation-in-the-americas for the organisations, and use the abbreviations like /amta for the event series

Can we have the nesting without adding it to the URL? So that we don't have `events/amta/amta2022`?

I'm thinking of something such as [`Building and research > Approaches > Neural machine translation`](https://machinetranslate.org/approaches), where the articles nested inside `Approaches` are only shown in the sidebar if we click on `Approaches`.

> I think either way we'll need to add `organisation:` to the front matter of event instances.

This information is already available in most events' front matter, except for the cases such as `NeTTT`, where there isn't an association related its organization.
SEO structured data need this data for rich results. Actually, I've worked on this last week as it may be related to #281.
![imagen](https://user-images.githubusercontent.com/33723604/182214881-dd07b12a-60fa-43e9-8ced-be5e7de24440.png)
